projects/user_publication/gscholar.py

```diff
@@ -0,0 +1,207 @@
+        cites_gen = self.scholarly.search_citedby(
+            pub_id, year_low=year_low, year_high=year_high
+        )
+        citations = []
+        cit_count = 0
+        while True:
+            try:
+                cited_pub = self._get_bib(self._get_next_cite(cites_gen))
+                citations.append(cited_pub)
+                cit_count += 1
+                logger.info(f"Got {len(cit_count)} / {no_citations}")
+            except StopIteration:
+                logger.info(f"End of iteration. Got {len(citations)} / {no_citations}")
+                return citations
+
+    @_handle_proxy_reload
+    def _get_next_cite(self, cites_gen):
+        return next(cites_gen)
+
+    @_handle_proxy_reload
+    def _get_bib(self, pub):
+        bibt = self.scholarly.bibtex(pub)
+        pub.update({"bibtex": bibt})
+        return pub
+
+    def get_authors(self, pub):
+        """returns a list of authors in a publication
+
+        Args:
+            pub (dict): scholarly return of the publication
+
+        Returns:
+            list: list of authors in ["firstname lastname",] format
+        """
+        bibt = pub['bibtex']
+        bibt = utils.decode_unicode_text(bibt)
+        match = re.search(r'author\s*=\s*\{([\w\s-,]+)\}', bibt)
+        if match:
+            authors_t = match.groups()[0]
+        else:
+            # this mostly never happens - if it does - log it
+            logger.info(f"cannot parse bibtex, so ignoring - {bibt}")
+            return
+        if "and" in authors_t:
+            authors = authors_t.split(" and ")
+        else:
+            authors = [authors_t]
+        parsed_authors = []
+        for author in authors:
+            parsed_authors.append(utils.parse_author(author))
+        return authors
+
+    def get_pub_model(self, pub):
+        return Publication(
+            title=utils.decode_unicode_text(pub["bib"]["title"]),
+            year=pub["bib"]["pub_year"],
+            author=" and ".join(self.get_authors(pub)),
+            entry_created_date=datetime.date.today(),
+            publication_type=utils.get_pub_type(pub["bib"]["ENTRYTYPE"]),
+            bibtex_source="{}",
+            added_by_username="admin",
+            forum=pub["bib"]["booktitle"],
+            link=pub["pub_url"],
+            source="gscholar",
+            status=Publication.STATUS_IMPORTED,
+        )
+
+
+def pub_import(
+    dry_run=True, year_low=2014, year_high=datetime.now().year
+):
+    """Returns Publication models of all publicationls that ref chameleon
+    and are not in database already
+    - Checks if there is a project associated with the publication
+
+    Args:
+        dry_run (bool, optional): Defaults to True.
+        year_low (int, optional): Defaults to 2014.
+        year_high (int, optional): Defaults to datetime.now().year.
+    """
+    gscholar = GoogleScholarHandler()
+    pubs = []
+    for chameleon_pub in ChameleonPublication.objects.filter(id=14):

```

Comment:
This filters for the one Chameleon publication that has ID `14`.

Reply:
Right. This was used for debugging